cli/azd/pkg/auth/manager.go

```diff
@@ -206,8 +206,13 @@ func (m *Manager) CredentialForCurrentUser(
 			return newCredentialFromClientSecret(tenantID, *currentUser.ClientID, *ps.ClientSecret)
 		} else if ps.ClientCertificate != nil {
 			return newCredentialFromClientCertificate(tenantID, *currentUser.ClientID, *ps.ClientCertificate)
-		} else if ps.FederatedToken != nil {
-			return newCredentialFromClientAssertion(tenantID, *currentUser.ClientID, *ps.FederatedToken)
+		} else if ps.FederatedAuth != nil {
+			if ps.FederatedAuth.Token != nil {
+				return newCredentialFromClientAssertion(tenantID, *currentUser.ClientID, *ps.FederatedAuth.Token)
+			} else if ps.FederatedAuth.TokenProvider != nil {
+				//nolint:lll
+				return m.newCredentialFromFederatedTokenProvider(tenantID, *currentUser.ClientID, *ps.FederatedAuth.TokenProvider)
```

Comment:
```suggestion
				return m.newCredentialFromFederatedTokenProvider(
				    tenantID, *currentUser.ClientID, *ps.FederatedAuth.TokenProvider)
```

Reply:
nit.  I just think we are abusing of `//nolint:lll` everywhere hahahah

Reply:
We could also just fold this condition into the above if:

```
} else if ps.FederatedAuth != nil && ps.FederatedAuth.Token != nil {
    ...
} else if ps.FederatedAuth != nil && ps.FederatedAuth.TokenProvider != nil {
    ...
}
```

which would let us save a tab-stop.

Reply:
@vhvb1989 Wasn't trying to abuse it. I indented the parameters and it read really off when compared to the rest of the function.. I then tried indenting other functions, and they looked pretty worse for wear 😂

Reply:
It's now indented. FYI, it was 3 characters off in the original implementation, and my personal experience was that a single line initialization would have read better, even on a laptop.